### PR TITLE
Allow multiple endpoints + headers

### PR DESCRIPTION
**Pull request by savoie**
**My use case:** Needing to scrape multiple different endpoints without running a separate exporter + config combo for each. I also needed to be able to pass HTTP headers with my requests (in my case, auth keys).

**Implementation details:** I think the examples I've changed in/added to the README speak for themselves; the new config format is a list of "modules", each of which has an endpoint, an (optional) header map, and a list of JSON -> metric mappings.

**Goal of this PR:** To provide a functional version with these features for anyone who wants to build it; to get feedback on whether this is a good addition to the exporter or should remain a fork. The changes probably belong in separate PRs, but the changes built on each other and I'm not certain this repo is still being maintained.

@kawamuray Any feedback appreciated. (I'm quite new to golang and may have written unidiomatic code.)

### CODE REVIEW

**Comment by GalaxyGorilla** [comment]
@kawamuray Can you pls check this PR? Having multiple endpoints quite an important feature for my usecase too and I would love to see this being merged 👍 .

File: jsonexporter/config.go [hunk 1]
```diff
@@ -6,42 +6,50 @@ import (
...
+    Endpoint string            `yaml:"endpoint"`
+    Headers  map[string]string `yaml:"headers"`
+    Mappings []*Mapping        `yaml:"mappings"`
 }
 
-func (config *Config) labelNames() []string {
-	labelNames := make([]string, 0, len(config.Labels))
-	for name := range config.Labels {
+type Mapping struct {
+	Name   string              `yaml:name`
```
Comment (kawamuray) [inline jsonexporter/config.go]: looks like it has been from the original code but tag values needed to be quoted by `"`?

File: jsonexporter/version.go [hunk 2]
```diff
@@ -1,3 +1,3 @@
 package jsonexporter
 
-const Version = "0.0.1"
+const Version = "0.1.1"
```
Comment (kawamuray) [inline jsonexporter/version.go]: I don't have specific versioning policy for this kind of small project but would be better to bump this to `1.0.0` to indicate breaking change?

File: README.md [hunk 3]
```diff
@@ -68,6 +70,31 @@ example_value_count{environment="beta",id="id-A"} 1
...
 ```
 
+Headers & Additional Endpoints
+=============
+An example configuration for multiple endpoints, with headers:
+
+```
+- endpoint: http://example.com/endpoint-1
+  headers:
+  - Content-Type: application/json
```
Comment (kawamuray) [inline README.md]: I think this YAML causes parsing error as [`ModuleConfig`](https://github.com/kawamuray/prometheus-json-exporter/pull/4/files#diff-719916233fcf8fcf593dcaef527bb211R11) declares `Headers` as `map[string]string` but these lines are interpreted as list of strings.
AFAIK multiple headers with same name is allowed in HTTP but considering practical use case I think it's okay to get rid of leading dash `-` and make these "map".

Comment (zmlpjuran) [inline README.md]: I think better example is:
headers:
- Accept: application/json

As you are not sending any content in Get request. Actually this is something I'm looking for. So, I like your change.

**Comment by zmlpjuran** [comment]
I'm not sure if the old example will work after this change at all. Perhaps it will be better to replace it.

**Comment by zmlpjuran** [comment]
Hi guys,
what is the progress of this PR?
@savoie Can I help you?

what is missing for this PR?
- reformat code
- bump version to 1.0.0
- update documentation
- deal with headers @kawamuray multiple header fields are allowed and I think it is OK to be complaint with HTTP/1.1 section 4.2 see https://www.w3.org/Protocols/rfc2616/rfc2616-sec4.html#sec4.2. So, maybe the list of string string is appropriate? OR it can stay simple as now and follow your suggestion. What is your opinion?

**Comment by VR6Pete** [comment]
Hey guys, are there any updates on this  ? would love to see support for multiple targets :)

**Comment by VR6Pete** [comment]
Hey @zmlpjuran and @savoie 

What’s the current status of this pull request? Are you close to having something to merge back in or have you forked this project under a different name?

**Pull request closed without merging**